airbyte-integrations/connectors/source-asana/source_asana/manifest.yaml

```diff
@@ -0,0 +1,2566 @@
+version: "0.44.0"
+
+definitions:
+  selector:
+    type: RecordSelector
+    extractor:
+      type: DpathExtractor
+      field_path: ["{{ parameters.extractor_path }}"]
+
+  oauth_authenticator:
+    type: OAuthAuthenticator
+    client_id: "{{ config['credentials']['client_id'] }}"
+    client_secret: "{{ config['credentials']['client_secret'] }}"
+    refresh_token: "{{ config['credentials']['refresh_token'] }}"
+    token_refresh_endpoint: "https://app.asana.com/-/oauth_token"
+    grant_type: refresh_token
+    access_token_name: access_token
+
+  pat_authenticator:
+    type: "BearerAuthenticator"
+    api_token: "{{ config['credentials']['personal_access_token'] }}"
+
+  selective_authenticator:
+    type: SelectiveAuthenticator
+    authenticator_selection_path: ["credentials", "option_title"]
+    authenticators:
+      "OAuth Credentials": "#/definitions/oauth_authenticator"
+      "PAT Credentials": "#/definitions/pat_authenticator"
+
+  requester:
+    type: HttpRequester
+    url_base: https://app.asana.com/api/1.0/
+    http_method: GET
+    authenticator:
+      $ref: "#/definitions/selective_authenticator" # Uncomment for development with low code
+      # $ref: "#/definitions/oauth_authenticator"         # Uncomment for developing with connector builder
+    error_handler:
+      type: CompositeErrorHandler
+      error_handlers:
+        - type: DefaultErrorHandler
+          response_filters:
+            - type: HttpResponseFilter
+              action: IGNORE
+              predicate: >-
+                {{ 'App Only access is not allowed' in response
+                }}
+              http_codes:
+                - 403
+                - 404
+                - 412
+              error_message: Resource is forbidden
+
+  token_paginator_base:
+    type: DefaultPaginator
+    page_token_option:
+      type: "RequestPath"
+      field_name: "sync"
+      inject_into: "request_parameter"
+    pagination_strategy:
+      type: CursorPagination
+      cursor_value: '{{ response.get("sync", {}) }}'
+      stop_condition: '{{ not response.get("sync", {}) }}'
+
+  paginator_base:
+    type: DefaultPaginator
+    page_token_option:
+      type: RequestOption
+      inject_into: request_parameter
+      field_name: offset
+    page_size_option:
+      type: RequestOption
+      field_name: limit
+      inject_into: request_parameter
+    pagination_strategy:
+      type: OffsetIncrement
+      page_size: 50
+
+  retriever:
+    type: SimpleRetriever
+    record_selector:
+      $ref: "#/definitions/selector"
+    paginator:
+      type: NoPagination
+    requester:
+      $ref: "#/definitions/requester"
+
+  base_stream:
+    type: DeclarativeStream
+    retriever:
+      $ref: "#/definitions/retriever"
+
+  projects_stream:
+    $ref: "#/definitions/base_stream"
+    name: "projects"
+    primary_key: "gid"
```

Comment:
Could you please move the `primary_key` param to the `base_stream` as it is the same for all streams except `events`?